csrc/scheduler/matmul_utils.cpp

```diff
@@ -449,13 +449,26 @@ std::string getMatmulRunTimeRejectReason(
 //  by the analysis.
 std::string getMatmulCompileTimeRejectReason(Fusion* fusion) {
   // The plan:
+  // 0. Check if the current CUDA device is supported
   // 1. Check if there is exactly one MmaOp or suitable mul sum pair
   // defined in the fusion.
   // 2. Check if inputs to the mma op or mul sum pair match any of
   // supported inputs layout
   // 3. Check if fusion represents expressions that are recognized by matmul
   // scheduler.
 
+  // #0
+  {
+    const auto device_prop = at::cuda::getCurrentDeviceProperties();
```

Comment:
We also have this utility here: https://github.com/NVIDIA/Fuser/blob/cdcb379d4e28dcb1359e6c673e5c5b162ece8414/csrc/scheduler/matmul_utils.cpp#L46-L68

This is used in `getMatmulHeuristics`:
https://github.com/NVIDIA/Fuser/blob/cdcb379d4e28dcb1359e6c673e5c5b162ece8414/csrc/scheduler/matmul_utils.cpp#L539-L543

These look like very similar use cases so we should preferably combine them.

Reply:
Additionally, this check is present in `getMatmulHeuristics` which is called by `computeHeuristics` in the matmul scheduler constructor. Do we need this check additionally in `getCompileTimeRejectReason`. Maybe this check should only be in `getCompileTimeRejectReason` instead of raising an error in `getMatmulHeuristics`.

Reply:
Ahh right, sorry I missed that is the same function being used.